### PR TITLE
[java][jersey] Remove double brace initialization and some more jersey cleanup

### PR DESCRIPTION
**Pull request by snago**
Some general cleanups in the Jersey 2 & 3 client generators.

Mainly remove double brace initialization in generated code, otherwise it's mostly about making the generated code more readable.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team as the change covers default generator

**Comment by snago** [comment]
I noticed that this PR and #14735 will have merge conflicts. I will happily rebase this if #14735 is merged first.

File: modules/openapi-generator/src/main/resources/Java/libraries/jersey2/ApiClient.mustache [hunk 1]
```diff
@@ -944,10 +972,11 @@ public class ApiClient{{#jsr310}} extends JavaTimeFormatter{{/jsr310}} {
   /**
    * Escape the given string to be used as URL query value.
    *
-   * @param str String
+   * @param object Any object to be turned into a String and escaped
    * @return Escaped string
    */
-  public String escapeString(String str) {
+  public String escapeString(Object object) {
+    String str = object.toString();
```
[the remark this answers is not in the document]
Comment (snago) [reply modules/openapi-generator/src/main/resources/Java/libraries/jersey2/ApiClient.mustache:979]: IIRC, the arguments were null-checked before calling this method (I moved `.toString()` from the call-site into the method).

But I'll take a closer look...
Any preference for return value if the argument is null? Empty string? Null?

[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/main/resources/Java/libraries/jersey2/ApiClient.mustache:979]: > Any preference for return value if the argument is null? Empty string? Null?

I think we should return the same "value" as before

[the remark this answers is not in the document]
Comment (snago) [reply modules/openapi-generator/src/main/resources/Java/libraries/jersey2/ApiClient.mustache]: I found another solution.
I've reverted the changes to this method and instead only apply `.toString()` to non-string path parameters.

**Comment by wing328** [comment]
please resolve the merge conflicts when you've time?

**Comment by snago** [comment]
Done! Rebased on top of current master.

**Comment by snago** [comment]
I just saw issue #14761 and realized this PR fixes that issue as I changed header, cookie and form parameters to LinkedHashMap so the will be sent in the same order as they are added.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java [hunk 2]
```diff
@@ -820,6 +820,11 @@ Map<String, Object> buildSupportFileBundle(List<OperationsMap> allOperations, Li
             bundle.put("hasServers", true);
         }
 
+        boolean hasOperationServers = allOperations != null && allOperations.stream()
+                .flatMap(om -> om.getOperations().getOperation().stream())
+                .anyMatch(o -> o.servers != null && !o.servers.isEmpty());
+        bundle.put("hasOperationServers", hasOperationServers);
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:827]: FYI @OpenAPITools/generator-core-team

**Comment by wing328** [comment]
@snago thanks again for your PR.  Did you test with your spec or use cases and everything is ok after the change?

**Comment by snago** [comment]
I have only tested this on the samples so far.
Having the samples was great during development. Make a small change to the template, run generate-samples.sh, see the changes, rinse, repeat.
At some point during this process I did tweak the samples a bit locally to test some edge case, but that's it.

I have two spec-files at work I could try but I don't expect them to cover anything more than what the samples doesn't already cover.

**Comment by wing328** [comment]
thanks for sharing more with me

> I have two spec-files at work

please test these spec files if you can 

if no further feedback from anyone, I'll merge this PR over the weekend.

**Comment by snago** [comment]
I've tried out the two spec-files at work, generating clients with 6.4.0 and with this branch and comparing the results.

Good thing I did because I found a bug, so **don't merge this yet!**
I'll get a fix in later today or tomorrow.

**Comment by wing328** [comment]
Please take your time. I'll put it on hold for the time being.

**Comment by snago** [comment]
That should do it! Now it works correctly on all spec files I've tried it on :)

Now I will also have these two work spec files to test with for future changes.

Speaking of, should I try to keep any future pull requests smaller? Or is it OK to group multiple somewhat related changes into one PR, like this? E.g. should the four first commits in this PR have been four different PRs?

**Comment by wing328** [comment]
> Speaking of, should I try to keep any future pull requests smaller?

defining the smaller the better as it's much easier to review small change and get it merged.

**Comment by wing328** [comment]
thanks for the PR. let's go ahead with this change.
